addons/addon-base/packages/serverless-go-build-tools/index.js

```diff
@@ -79,6 +92,45 @@ class GoBuildTools {
       return;
     }
 
+    // Test out Go command
+    try {
+      execSync('go version');
```

Comment:
What would happen is go is installed but the go build failed for other reason? Will the deployment script success despite the failure?

Reply:
Yes, it would have done that to begin with as well. I suppose I could check the error in the second try-catch (at line 176) to make sure the error is missing go command. Would you like me to implement that?

Reply:
Yes, I think we can move what's in the catch block to line 176 directly, and catch go not installed and go build failed altogether. And we can use a more generic error message and remediation recommendation  so it takes care of both scenarios.

Reply:
We did it this way for the prompts to be synchronous.

Reply:
We did the prompting first so that we can have the prompts be synchronous. In the second try-catch, they are nested within a few operations so the prompt prints wrong.